cypress/e2e/opening-hours-editor.cy.ts

```diff
@@ -0,0 +1,298 @@
+  }
+  if (end) {
+    cy.getBySel("opening-hours-editor-form-end-time").focus().type(end);
+  }
+};
+
+const submitOpeningHourForm = () => {
+  cy.getBySel("opening-hours-editor-form-submit").click();
+};
+
+const checkOpeningHoursAdmin = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  return cy
+    .getBySel("opening-hours-editor-event-content")
+    .should("be.visible")
+    .and("contain", openingHourCategory)
+    .and("contain", `${start} - ${end}`);
+};
+
+const checkOpeningHoursPage = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  cy.getBySel("opening-hours-week-list")
+    .should("be.visible")
+    .and("contain", openingHourCategory)
+    .and("contain", `${start} - ${end}`);
+};
+
+const checkOpeningHoursNotPresentInPage = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  cy.getBySel("opening-hours-week-list")
+    .should("be.visible")
+    .should("not.contain", openingHourCategory)
+    .should("not.contain", `${start} - ${end}`)
+    .contains("The library is closed this day");
+};
+
+const createOpeningHour = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  visitOpeningHoursPageAdmin();
+  navigateToMonthViewAdmin();
+  selectTodayFromMonthViewAdmin();
+  fillOpeningHourForm({ openingHourCategory, start, end });
+  submitOpeningHourForm();
+  checkOpeningHoursAdmin({ openingHourCategory, start, end });
+  visitOpeningHoursPage();
+  checkOpeningHoursPage({ openingHourCategory, start, end });
+};
+
+const createOpeningHourInNextWeek = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  visitOpeningHoursPageAdmin();
+  navigateToNextWeekAdmin();
+  selectTimeOnThursdayFromWeekView({ start });
+  fillOpeningHourForm({ openingHourCategory, end });
+  cy.getBySel("opening-hours-editor-form-start-time").should(
+    "have.attr",
+    "value",
+    start
+  );
+  submitOpeningHourForm();
+  visitOpeningHoursPage();
+  checkOpeningHoursNotPresentInPage({
+    openingHourCategory,
+    start,
+    end,
+  });
+  cy.getBySel("opening-hours-next-week-button").click();
+  checkOpeningHoursPage({ openingHourCategory, start, end });
+};
+
+const updateOpeningHour = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  // Assume that the event is already created and is visible
+  visitOpeningHoursPageAdmin();
+  navigateToMonthViewAdmin();
+  cy.getBySel("opening-hours-editor-event-content")
+    .contains(openingHourCategory)
+    .click();
+  fillOpeningHourForm({ start, end });
+  submitOpeningHourForm();
+  checkOpeningHoursAdmin({ openingHourCategory, start, end });
+  visitOpeningHoursPage();
+  checkOpeningHoursPage({ openingHourCategory, start, end });
+};
+
+const deleteOpeningHour = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  createOpeningHour({
+    openingHourCategory,
+    start,
+    end,
+  });
```

Comment:
Please consider removing `createOpeningHour()` from `deleteOpeningHour()`.

I do not see a reason to have this in the same function. I do not see a reason it should be related. If we remove it then the delete test will look like the update test.